cores/esp8266/flash_hal.cpp

```diff
@@ -41,7 +41,7 @@ int32_t flash_hal_read(uint32_t addr, uint32_t size, uint8_t *dst) {
 }
 
 int32_t flash_hal_write(uint32_t addr, uint32_t size, const uint8_t *src) {
-    optimistic_yield(10000);
+    optimistic_yield(1000);
```

Comment:
For here and the hal_read I'm not sure the original thinking behind the yield.  These operations are fixed duratin and IRQs/etc. are disabled 100% during their operation, so nothing happens during the actual transfer.  The transfer is, however, <1ms in time.

Given that, I'm not sure I'd increase the frequency of the yield in these two calls...